src/hipscat/pixel_math/box_filter.py

```diff
@@ -0,0 +1,146 @@
+from __future__ import annotations
+
+from typing import List, Tuple
+
+import astropy.units as u
+import healpy as hp
+import numpy as np
+from astropy.coordinates import Angle
+
+from hipscat.pixel_math import HealpixPixel
+from hipscat.pixel_math.filter import get_filtered_pixel_list
+from hipscat.pixel_math.polygon_filter import SphericalCoordinates
+from hipscat.pixel_tree.pixel_tree import PixelTree
+from hipscat.pixel_tree.pixel_tree_builder import PixelTreeBuilder
+
+
+def filter_pixels_by_box(
+    pixel_tree: PixelTree, ra: Tuple[float, float] | None, dec: Tuple[float, float] | None
+) -> List[HealpixPixel]:
+    """Filter the leaf pixels in a pixel tree to return a partition_info dataframe
+    with the pixels that overlap with a right ascension or the declination region.
+    Only one of ra and dec should be set, otherwise one should use polygonal search.
```

Comment:
Is this line true? I thought using both ra and dec was a totally find thing to do, and the polygon search would return something different because it connects the vertices with great circles?

Reply:
You're right @delucchi-cmu, I guess I mistakenly left this behind!